### PR TITLE
feat: migrate mocha

### PR DESCRIPTION
**Pull request by ericjeker**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -33,7 +33,8 @@
     "format": "prettier --write .",
     "jsdoc": "gulp jsdoc",
     "lint": "echo 'TODO eslint'",
-    "test": "mocha test/*.*.js"
+    "test": "mocha --require ts-node/register ./test/*.spec.ts",
+    "test:js": "mocha test/*.*.js"
```
Comment (clarkerican) [inline package.json:37]: I think you're also going to want to add this new command to the Github Testing CI Action (https://github.com/faker-js/faker/blob/main/.github/workflows/ci.yml) to ensure the tests still run during CI :)

Comment (ericjeker) [reply package.json:37]: I was thinking we could use this branch to migrate the tests in full. Not sure the other contributors wanted to move forward with that. But we can also run both the JS tests and TS tests, so we can PR on main instead, and just increase the number of tests?

Comment (clarkerican) [reply package.json:37]: It might be nice to just branch off main, that may prevent us from losing tests from other PRs that may concurrently add.

**Comment by Shinigami92** [comment]
I will close this because we will directly migrate the tests to vitest #235

**Pull request closed without merging**